addons/sale/models/sale.py

```diff
@@ -526,8 +526,6 @@ def action_invoice_create(self, grouped=False, final=False):
                 if line.display_type == 'line_section':
                     pending_section = line
                     continue
-                if float_is_zero(line.qty_to_invoice, precision_digits=precision):
```

Comment:
you can't delete this. you will need to test line type. Also why are notes so special? Why not headings?